kratos/geometries/geometry.h

```diff
@@ -572,13 +568,123 @@ class Geometry
+    {
+        return mPoints(GetVectorIndexFromMatrixIndices(GetNumberOfPointsU(), IndexU, IndexV));
+    }
+
+    /// Get pointer of point at u'th, v'th position
+    PointPointerType pGetPoint(const IndexType IndexU, const IndexType IndexV)
+    {
+        return mPoints(GetVectorIndexFromMatrixIndices(GetNumberOfPointsU(), IndexU, IndexV));
+    }
+
+    /// Get pointer of point at u'th, v'th position
+    TPointType const& GetPoint(const IndexType IndexU, const IndexType IndexV) const
+    {
+        return mPoints[GetVectorIndexFromMatrixIndices(GetNumberOfPointsU(), IndexU, IndexV)];
+    }
+
+    /// Get pointer of point at u'th, v'th position
+    TPointType& GetPoint(const IndexType IndexU, const IndexType IndexV) {
+        return mPoints[GetVectorIndexFromMatrixIndices(GetNumberOfPointsU(), IndexU, IndexV)];
+    }
+
+    /// Get pointer of point at u'th, v'th, w'th position
+    const PointPointerType pGetPoint(const IndexType IndexU, const IndexType IndexV, const IndexType IndexW) const
+    {
+        return mPoints(GetVectorIndexFromTensorIndices(GetNumberOfPointsU(), GetNumberOfPointsV(), IndexU, IndexV, IndexW));
+    }
+
+    /// Get pointer of point at u'th, v'th, w'th position
+    PointPointerType pGetPoint(const IndexType IndexU, const IndexType IndexV, const IndexType IndexW)
+    {
+        return mPoints(GetVectorIndexFromTensorIndices(GetNumberOfPointsU(), GetNumberOfPointsV(), IndexU, IndexV, IndexW));
+    }
+
+    /// Get pointer of point at u'th, v'th, w'th position
+    TPointType const& GetPoint(const IndexType IndexU, const IndexType IndexV, const IndexType IndexW) const
+    {
+        return mPoints[GetVectorIndexFromTensorIndices(GetNumberOfPointsU(), GetNumberOfPointsV(), IndexU, IndexV, IndexW)];
+    }
+
+    /// Get pointer of point at u'th, v'th, w'th position
+    TPointType& GetPoint(const IndexType IndexU, const IndexType IndexV, const IndexType IndexW) {
+        return mPoints[GetVectorIndexFromTensorIndices(GetNumberOfPointsU(), GetNumberOfPointsV(), IndexU, IndexV, IndexW)];
+    }
+
+    /// Number of Points in direction u
+    virtual SizeType GetNumberOfPointsU() const {
+        return mPoints.size();
+    }
+
+    /// Number of Points in direction v
+    virtual SizeType GetNumberOfPointsV() const {
+        return 1;
+    }
+
+    /// Number of Points in direction w
+    virtual SizeType GetNumberOfPointsW() const {
+        return 1;
+    }
+
+    /*
+    * @brief Computes a vector index from two matrix indicies.
+    * @return index within vector
+    */
+    static constexpr inline IndexType GetVectorIndexFromMatrixIndices(
+        const SizeType NumberU, const SizeType NumberV,
+        const IndexType IndexU, const IndexType IndexV) noexcept
+    {
+        return IndexV * NumberU + IndexU;
+    }
+
+    /*
+    * @brief Computes a vector index from three tensor indicies.
+    * @return index within vector
+    */
+    static constexpr inline IndexType GetVectorIndexFromTensorIndices(
```

Comment:
I think Tensor is not the best term. But one understands what is meant.

Reply:
In ANurbs I call it `single`/`double`/`triple` index (last one is not in use). But that's not really critical. I think the behaviour on irregular meshes is more important.

Reply:
btw: what is the reason to put this function into the geometry base class?

Reply:
Actually this comes from the irregular meshes (triangles), either we derive this function there, or the GetPoint functions. If it will be the GetPoint functions I totally agree that we need to extract this in an external file.